app/api/routes/product.py

```diff
@@ -0,0 +1,138 @@
+from decimal import Decimal
+from sqlalchemy import asc, desc
+
+router = APIRouter()
+
+
+@router.post(
+    "/products/", response_model=ProductResponse, status_code=status.HTTP_201_CREATED
+)
+def create_product(product_data: ProductCreate, db: Session = Depends(get_db)):
+    validator = ProductValidator(db)
+
+    try:
+        validator.validate_unique_name(product_data.name)
+        validator.validate_product_data(product_data)
+
+        new_product = Product(**product_data.dict())
+
+        db.add(new_product)
+        db.commit()
+        db.refresh(new_product)
+
+        return new_product
+    except InvalidProductDataException as e:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=e.detail)
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e)
+        )
+
+
+@router.get("/products/{product_id}", response_model=ProductResponse)
+def get_product(product_id: UUID, db: Session = Depends(get_db)):
+    service = ProductService(db)
+    try:
+        return service.get_product_by_id(product_id)
+    except ProductNotFoundException as e:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=str(e),
+        )
+
+
+@router.put("/products/{product_id}", response_model=ProductResponse)
+def update_product(
+    product_id: UUID, product_data: ProductUpdate, db: Session = Depends(get_db)
+):
+    validator = ProductValidator(db)
+    service = ProductService(db)
+
+    try:
+        existing_product = service.get_product_by_id(product_id)
+        if not existing_product:
+            raise ProductNotFoundException(product_id)
+
+        if product_data.name and product_data.name != existing_product.name:
+            validator.validate_unique_name(product_data.name)
+
+        validator.validate_product_data(product_data)
+
+        updated_product = service.update_product(product_id, product_data)
+
+        return updated_product
+    except ProductNotFoundException as e:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=str(e),
+        )
+    except InvalidProductDataException as e:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=e.detail)
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=str(e),
+        )
+
+
+@router.delete("/products/{product_id}", status_code=status.HTTP_204_NO_CONTENT)
+def delete_product(product_id: UUID, response: Response, db: Session = Depends(get_db)):
+    service = ProductService(db)
+    try:
+        service.delete_product(product_id)
+        response.status_code = status.HTTP_204_NO_CONTENT
+    except ProductNotFoundException as e:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=str(e),
+        )
+
```

Comment:
General exception

Reply:
Please address this

Reply:
Done, creating global exception handler for every routes . for unexpected behaviour .